enhancements/update/eus-upgrades-mvp.md

```diff
@@ -0,0 +1,368 @@
+should provide some mechanism to rank upgraded nodes as slightly higher priority
+in scheduler configuration. Perhaps a timestamp of configuration generation that
+the scheduler then biases toward nodes with the greatest timestamps as a last in
+chain priority.
+
+We should consider enabling the rescheduler as well to ensure that when the
+upgrade completes we rebalance so that the last node ends up with proportionate
+workload.
+
+#### ??? - Provide a hosted graph viewer which will assist disconnected admins in mirroring all content between 4.6 and 4.10
+
+Assume that 4.6.eus-z -> 4.7.z -> 4.8.z -> 4.9.z -> 4.10.z are all in the eus-4.10
+channel providing a chain of upgrades that eventually lead to 4.10. We should build
+a graph viewer / solver somewhere which allows an admin of a 4.6 cluster hoping
+to upgrade to 4.10 to discover all content which must be mirrored between their
+current version and their eventual 4.10 target.
+
+In general the graph viewer should provide a complete path from the cluster's
+current version to the latest .z in the selected channel assuming such a path exists.
+If someone were to provide cluster version 4.4.3 but select the stable-4.6 channel
+no such path exists and it should indicate that, if they were to select stable-4.5
+it would provide the upgrade path of 4.4.3 to 4.4.29 to 4.5.24 which is the latest 4.5.z.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The main caveat of what's outlined here is that it does not reduce reboots
+which, for many reasons, is highly desired by some admins. The rationale for not
+addressing that in this enhancement is that this enhancement targets
+improvements which benefit all upgrades.
+
+The epic / user story to backport API removal notifications to 4.6 assumes that
+we will require anyone upgrading along an EUS specific upgrade path to update to
+a 4.6 version which is specific to EUS. This is necessary because only those who
+follow the EUS upgrade path have 4.10 as their next logical target version. We
+should not alert non EUS clusters of removals in 4.10 until they've upgraded to
+a later minor version.
+
+As with all upgrades, new features enabled by interim versions are not expected
+enabled during the upgrade path. This means that for instance new MachineConfig
+features enabling complex root devices while introduced in 4.7 may not be utilized
+until the cluster has completed its upgrade.
+
+### Risks and Mitigations
+
+The improvements here do not seem to introduce risk unto themselves. However
+failure to implement some of these such as the version skew constraints would
+impose significant risk to supportability in clusters which may upgrade more
+rapidly than is safe to do so.
+
+Additionally, the timeframe for final delivery extends three releases into the
+future. This allows for design iteration during the 4.8 release cycle with
+implementation focused on the 4.9 and 4.10 release cycles.
+
```

Comment:
I miss the risk of upstream making our efforts hard or infeasible.

Reply:
@sttts specifically around items outlined here, mainly api removals, or something else?

I do intend to call out upstream's apparent lack of kubelet N-2 version skew testing despite the stated policy that it's supported.

Reply:
> @sttts specifically around items outlined here, mainly api removals, or something else?

afaict, all of them. We only have this amount of time we can juggle between upstream and downstream. It happened in the past that we missed changes upstream that caused us pain. They are rare, but they still can happen without our sufficient upstream presence to mitigate them.

Reply:
I've added a paragraph about EUS to EUS not being something that upstream considers. PTAL.